app/src/main/kotlin/com/aliucord/manager/ui/viewmodel/InstallViewModel.kt

```diff
@@ -350,69 +357,22 @@ class InstallViewModel(
         }
     }
 
-    var currentCategory: StepCategory? by mutableStateOf(StepCategory.APK_DL)
-
-    // Shared steps
-    val appIconPatch by mutableStateOf(Step(
-        "Patching app icons",
-        Status.QUEUED
-    ))
-
-    val manifestPatch by mutableStateOf(Step(
-        "Patching apk manifests",
-        Status.QUEUED
-    ))
-
-    val dexPatch by mutableStateOf(Step(
-        "Adding aliu dex into apk",
-        Status.QUEUED
-    ))
-
-    val libPatch by mutableStateOf(Step(
-        "Replacing libraries",
-        Status.QUEUED
-    ))
-
-    val signApk by mutableStateOf(Step(
-        "Signing apks",
-        Status.QUEUED
-    ))
-
-    val installingApk by mutableStateOf(Step(
-        "Installing apks",
-        Status.QUEUED
-    ))
-
-    // RN steps
-    val baseApkDl by mutableStateOf(Step(
-        "Downloading base apk",
-        Status.QUEUED
-    ))
-
-    val librariesApkDl by mutableStateOf(Step(
-        "Downloading libraries apk",
-        Status.QUEUED
-    ))
-
-    val localeApkDl by mutableStateOf(Step(
-        "Downloading locale apk",
-        Status.QUEUED
-    ))
-
-    val resourceApkDl by mutableStateOf(Step(
-        "Downloading resource apk",
-        Status.QUEUED
-    ))
-
-    val hermesDl by mutableStateOf(Step(
-        "Downloading hermes & c++ runtime library",
-        Status.QUEUED
-    ))
-
-    val aliuNativeDl by mutableStateOf(Step(
-        "Downloading AliucordNative library",
-        Status.QUEUED
-    ))
+    enum class Steps {
+        PATCH_APP_ICON,
+        PATCH_MANIFEST,
```

Comment:
Can't step have a property of the group it belongs to
So that you can't have something like PATCH_LIBS as part of downloading group